internal/topic/topicreaderinternal/reader.go

```diff
@@ -68,11 +68,11 @@ func NewReader(
 	consumer string,
 	readSelectors []topicreadercommon.PublicReadSelector,
 	opts ...PublicReaderOption,
-) (Reader, error) {
+) (Reader, *context.Context, error) {
```

Comment:
Why you want return context pointer?
It is very uncommon way.

Reply:
What about add method to get current context to stream reader, reader reconnector, etc?
And it must not be context pointer - for prevent races while context changing.